File: pyomo/with_symmetry_elimination_constraints/cp_2_isosceles_x_ordering_noinit.py
# NLP written by GAMS Convert at 03/17/24 14:47:59
#
# Equation counts
#     Total        E        G        L        N        X        C        B
#         8        0        7        1        0        0        0        0
#
# Variable counts
#                  x        b        i      s1s      s2s       sc       si
#     Total     cont   binary  integer     sos1     sos2    scont     sint
#         5        5        0        0        0        0        0        0
# FX      0
#
# Nonzero counts
#     Total    const       NL
#        21       16        5
#
# Reformulation has removed 1 variable and 1 equation

from pyomo.environ import *

model = m = ConcreteModel()

m.x1 = Var(within=Reals, bounds=(0,1), initialize=0)
m.x2 = Var(within=Reals, bounds=(0,1), initialize=0)
m.x3 = Var(within=Reals, bounds=(0,1), initialize=0)
m.x4 = Var(within=Reals, bounds=(0,1), initialize=0)
m.x5 = Var(within=Reals, bounds=(None,None), initialize=0)

m.obj = Objective(sense=maximize, expr= m.x5)

m.e1 = Constraint(expr= (m.x1 - m.x2)**2 + (m.x3 - m.x4)**2 - (2 * m.x5)**2
    >= 0)
m.e2 = Constraint(expr= m.x1 - m.x2 <= 0)
m.e3 = Constraint(expr= m.x1 - m.x5 >= 0)
m.e4 = Constraint(expr= m.x2 - m.x5 >= 0)
m.e5 = Constraint(expr= m.x3 - m.x5 >= 0)
m.e6 = Constraint(expr= m.x4 - m.x5 >= 0)
m.e7 = Constraint(expr= -0.7071067811865475 * m.x1 - 0.7071067811865475 * m.x3
    - m.x5 >= -0.7071067811865475)
m.e8 = Constraint(expr= -0.7071067811865475 * m.x2 - 0.7071067811865475 * m.x4
    - m.x5 >= -0.7071067811865475)

File: jump/with_symmetry_elimination_constraints/cp_2_isosceles_x_ordering_noinit.jl
# NLP written by GAMS Convert at 03/17/24 14:47:55
#
# Equation counts
#     Total        E        G        L        N        X        C        B
#         8        0        7        1        0        0        0        0
#
# Variable counts
#                  x        b        i      s1s      s2s       sc       si
#     Total     cont   binary  integer     sos1     sos2    scont     sint
#         5        5        0        0        0        0        0        0
# FX      0
#
# Nonzero counts
#     Total    const       NL
#        21       16        5
#
# Reformulation has removed 1 variable and 1 equation

using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, 0 <= x1 <= 1, start=0)
@variable(m, 0 <= x2 <= 1, start=0)
@variable(m, 0 <= x3 <= 1, start=0)
@variable(m, 0 <= x4 <= 1, start=0)
@variable(m, x5, start=0)

@objective(m, Max, x5)

@NLconstraint(m, e1, (x1 - x2)^2 + (x3 - x4)^2 - (2 * x5)^2 >= 0)
@constraint(m, e2, x1 - x2 <= 0)
@constraint(m, e3, x1 - x5 >= 0)
@constraint(m, e4, x2 - x5 >= 0)
@constraint(m, e5, x3 - x5 >= 0)
@constraint(m, e6, x4 - x5 >= 0)
@constraint(m, e7, -0.7071067811865475 * x1 - 0.7071067811865475 * x3 - x5
    >= -0.7071067811865475)
@constraint(m, e8, -0.7071067811865475 * x2 - 0.7071067811865475 * x4 - x5
    >= -0.7071067811865475)